core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -0,0 +1,100 @@
+package io.grpc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * TODO: Insert description here. (generated by notcarl)
+ */
+public final class MessageEncoding {
+
+  /**
+   * Special sentinel codec indicating that no compression should be used.  Users should use
+   * reference equality to see if compression is disabled.
+   */
+  public static final Codec NONE = new Codec() {
+    @Override
+    public InputStream decompress(InputStream is) throws IOException {
+      return is;
+    }
+
+    @Override
+    public String getMessageEncoding() {
+      return null;
+    }
+
+    @Override
+    public OutputStream compress(OutputStream os) throws IOException {
+      return os;
+    }
+  };
+
+
+  /**
+   * Represents a message compressor.
+   */
+  public interface Compressor {
```

Comment:
Thinking out loud:

One of the ideas I had for how this would work would be to add methods to different compressors.  For example, Gzip could have a setCompressionLevel method on it that would be specific to it (and perhaps not snappy, or other compression type).  Using such a method would be specific to compressing, but not decompressing, so it would look odd.  Something I want to avoid:
```
// This doesn't do what you think it does
codec.setCompressionLevel(Level.FAST).decode(message);
```

Specifically, it is possible to imagine wanting to set encode or decode specific methods that would not apply to the other.  While not the end of the world, it would add to the wartiness of the API.  

It would also make testing /slightly/ easier since framers and deframers only care about implementing one of those methods.